alf/environments/alf_wrappers.py

```diff
@@ -465,3 +481,285 @@ def reward_spec(self):
+            action=self._action_spec)
+
+    @staticmethod
+    def load(load_fn, environment_name, **kwargs):
+        """
+        Args:
+            load_fn (Callable): function used to construct the environment for
+                each tasks. It will be called as ``load_fn(env_name, **kwargs)``
+            environment_name (list[str]): list of environment names
+            kwargs (**): arguments passed to load_fn
+        """
+        envs = []
+        for name in environment_name:
+            envs.append(load_fn(name, **kwargs))
+        return MultitaskWrapper(envs, environment_name)
+
+    @property
+    def num_tasks(self):
+        return len(self._envs)
+
+    @property
+    def task_names(self):
+        return self._task_names
+
+    def observation_spec(self):
+        return self._observation_spec
+
+    def action_spec(self):
+        return self._action_spec
+
+    def reward_spec(self):
+        return self._reward_spec
+
+    def env_info_spec(self):
+        return self._env_info_spec
+
+    def get_num_tasks(self):
+        return len(self._envs)
+
+    def _reset(self):
+        time_step = self._envs[self._current_env_id].reset()
+        return time_step._replace(
+            prev_action=OrderedDict(
+                task_id=self._current_env_id, action=time_step.prev_action))
+
+    def _step(self, action):
+        self._current_env_id = action['task_id']
+        action = action['action']
+        assert self._current_env_id < len(self._envs)
+        time_step = self._envs[self._current_env_id].step(action)
+        return time_step._replace(
+            prev_action=OrderedDict(
+                task_id=self._current_env_id, action=time_step.prev_action))
+
+    def __getattr__(self, name):
+        if name.startswith('_'):
+            raise AttributeError(
+                "attempted to get missing private attribute '{}'".format(name))
+        return getattr(self._envs[self._current_env_id], name)
+
+    def seed(self, seed):
+        for env in self._envs:
+            env.seed(seed)
+
+
+@gin.configurable(blacklist=['env'])
+class CurriculumWrapper(AlfEnvironmentBaseWrapper):
+    """A wrapper to provide automatic curriculum task selection.
+
+    The probability of a task being chosen is based on its recent progress in
+    terms of episode reward. A task will be chosen more often if its episode
+    reward increases faster than other tasks.
+
+    The progress of a task is defined as the difference between its current score
+    and its past score.
+    """
+
+    def __init__(self,
+                 env,
+                 progress_favor=10.0,
+                 current_score_update_rate=1e-3,
+                 past_score_update_rate=5e-4,
+                 warmup_period=100):
+        """
+        env (AlfEnvironment): environment to be wrapped. It needs to be batched.
+        progress_favor (float): how much more likely to choose the environment with the
```

Comment:
This number is multiplied to all progress ratios. So this description is not accurate.

Reply:
This is indeed correct. The probability of the fastest one being sampled will be  ``progress_favor`` times more than the one with a task of 0 or negative progress (if there is one).